src/components/shoppingCart/paymentbar/PaymentSum.jsx

```diff
@@ -0,0 +1,64 @@
+import React, { useState, useEffect } from "react";
+import Container from "react-bootstrap/Container";
+import Col from "react-bootstrap/Col";
+import Row from "react-bootstrap/Row";
+import Button from "react-bootstrap/Button";
+import "./PaymentSum.scss";
+
+// TODO(emrerdem1): To get these datas from Redux.
+const SHIPPING_COST = 10;
+const AMOUNT_TO_FREE_SHIPPING = 150;
+const TEMPORARY_SUBTOTAL = 160;
+
+const PaymentSum = () => {
+  const [shippingCost] = useState(SHIPPING_COST);
+  const [isShippingFree, setIsShippingFree] = useState(false);
```

Comment:
This is not a comment that requires changes, but it is an educational one. Although for this app you don't have to do it and it's probably not possible because you use Firebase, this is exactly an example of a more complex feature that requires a server (perhaps Firebase does have an offering that does this, like Firebase functions or something).

One thing to consider is that someone always has access to client-side code. So for example, whether shipping is free or not is currently controlled by the JS-side. Well, using the console, I could change the js-side however I want. In fact, I can change the cost, shipping cost, and subtotal however I want, so without server-side validation, I can actually just use the console (it will be a bit obsfucated because it's react code but in the end there is definitely a way) to simply change all these values to 0 and buy everything from your site for free (it depends on how your payment processing is implemented of course). But the idea here is, these should be for display-only; the client should not control the actual amount that is sent to be charged.

Usually, you would retrieve these values from the DB on the server-side, the server would send them to the client to show, and when you checkout, the server would query the DB again for the values, perhaps based on the product ID (it would not get the values from the client-side).

Food for thought.

Reply:
Bunu hiç düşünmemiştim, harbiden de dediğin gibi; son derece tehlikeli. Lesson is taken.  💯